datanode/api/trading_data_v2.go

```diff
@@ -2218,7 +2218,7 @@ func (t *tradingDataServiceV2) ListNodeSignatures(ctx context.Context, req *v2.L
 
 	sigs, pageInfo, err := t.notaryService.GetByResourceID(ctx, req.Id, pagination)
 	if err != nil {
-		return nil, apiError(codes.NotFound, err)
+		return nil, fmt.Errorf("could not retrieve resource: %w", err)
```

Comment:
I still don’t think this is correct, and isn’t what I said in DMs. It is not necessarily an error that there is no result for a resource request, so no error should be passed back. An empty list is sufficient.

What might be an error is if the resource ID could never match (e.g. if I search for ‘🍝’), in which case an error specifying that the ID is an invalid format *could* be an error.

Reply:
The resource ID checking takes place in the GetByResourceID method as it handles the conversion from string to resource ID, and returns an error if an invalid resource ID is provided. If you pass a valid Hex ID and no results are found, then you will not get an error, just an empty list. 

GetByResourceID has to return an error and it needs to be handled here because there are other database issues that may occur that could return an error (e.g. connectivity), and I believe it's appropriate for the API to return those errors which this should do.

Reply:
Thanks for taking the time to explain it to me @guoguojin - I got too big for my boots and tried to review some Go.